src/cryptonote_core/uptime_proof.cpp

```diff
@@ -91,6 +93,9 @@ Proof::Proof(
                 tools::make_from_guts<eth::bls_public_key>(proof.require<std::string_view>("bk"sv));
         pop_bls =
                 tools::make_from_guts<eth::bls_signature>(proof.require<std::string_view>("bp"sv));
+        if (proof.skip_until("gh")) {
+            git_version = proof.consume_string();
```

Comment:
This should have a size check on it (perhaps 100 bytes, or maybe somewhat less than that?) with a rejection of the proof if it exceeds it + a corresponding size limit in generated values (just in case the version tag happens to be longer).  Otherwise the gossip mechanism would open up an amplification attack: I feed you a proof with a huge `gh` value, and then you propagate that to all of your peers, and they propagate it, etc.